pkgs/by-name/lm/lm4flash/package.nix

```
@@ -0,0 +1,48 @@
+{
+  fetchFromGitHub,
+  stdenv,
+  libusb1,
+  pkg-config,
+  lib,
+  writeText,
+}:
+let
+  stellaris-udev-rules = writeText "61.stellpad.rules" ''
+    SUBSYSTEM=="usb", ATTRS{idVendor}=="1cbe", ATTRS{idProduct}=="00fd", MODE="0666"
+  '';
+in
+stdenv.mkDerivation rec {
+  pname = "lm4flash";
+  version = "0.1.3";
+  src = fetchFromGitHub {
+    owner = "utzig";
+    repo = "lm4tools";
+    rev = "v${version}";
+    hash = "sha256-ZjuCH/XjQEgg6KHAvb95/BkAy+C2OdbtBb/i6K30+uo=";
+  };
+  sourceRoot = "${src.name}/lm4flash";
+  buildInputs = [
+    libusb1
+    pkg-config
+  ];
```

Dependencies should be separated into categories based on when they are supposed be used by the program.
Build-time only stuff goes into `nativeBuildInputs` and runtime stuff usually goes into `buildInputs`.

This distinction is needed when cross compiling programs, but it's useful in other ways too.

By default there is no difference between the `nativeBuildInputs` and `buildInputs`, but if you set `strictDeps = true` (which is recommended as a default nowadays) they will do separate things:

Programs listed in `nativeBuildInputs` will be put into the `PATH`, while programs listed in `buildInputs` will be able to be linked to and be `#include`-ed.

```suggestion

  strictDeps = true;

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = [
    libusb1
  ];
  
```